### ISSUE TITLE
SIM113 false positive with async for loops

### ISSUE THREAD

**Issue by adrienball** [astral-sh/ruff#9995]
### Steps to reproduce
- Create a debug.py file with the following content:
```python
async def my_async_gen():
    for i in range(5):
        yield i


async def my_async_func():
    idx = 0
    async for v in my_async_gen():
        idx += 1
        print(f"{idx}: {v}")
```
- Run `ruff check debug.py`

### Expected behavior
No error

### Actual behavior
```console
debug.py:9:9: SIM113 Use `enumerate()` for index variable `idx` in `for` loop
Found 1 error.
```

### Environment
```console
> ruff --version
ruff 0.2.1
```

**Comment by charliermarsh** [comment]
Ahh, nice catch!
